### PR TITLE
Fixed GPS location not updating. Added Halo circle to show location accuracy

### LINKED ISSUE #418: GPS Marker not moving
GPS user marker not moving on the map.

### PR DESCRIPTION
**Pull request by OSPFNeighbour**
Fixed #418 
Fixed missing keys on LocationMarker

Added a circle around the user marker to show location accuracy.

Added tests for new component.

screenshot 2018-05-11 11 24 27

### CODE REVIEW

File: client/src/components/MapMarker/MyLocationMarker.js [hunk 1]
```diff
@@ -19,6 +18,7 @@ const MyLocationMarker = ({ myPosition }) => {
 
   return (
     <Marker
+      key={(myPosition.longitude + myPosition.latitude).toString()}
```
Comment (sdunster) [inline client/src/components/MapMarker/MyLocationMarker.js]: I don't think this should change as the marker moves. I suggest using a static string like "my-location" or even add the user's ID number or something.

File: client/src/components/MapMarker/AccuracyHalo.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Circle } from 'react-native-maps';
+
+const AccuracyHalo = ({ myPosition }) => {
+  if
```
Comment (sdunster) [inline client/src/components/MapMarker/AccuracyHalo.js]:
```
if (
  !myPosition ||
  !myPosition.latitude ||
  !myPosition.longitude ||
  !myPosition.accuracy
) return null;
```

File: client/src/components/MapMarker/__tests__/__snapshots__/AccuracyHalo.js.snap [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`renders correctly 1`] = `null`;
```
Comment (sdunster) [inline client/src/components/MapMarker/__tests__/__snapshots__/AccuracyHalo.js.snap]: check your logic - both of these render to null which feels like a bug

File: client/src/components/MapMarker/MyLocationMarker.js (hunk 1 above)
Comment (OSPFNeighbour) [reply client/src/components/MapMarker/MyLocationMarker.js]: This was a work around for a 'bug' in maps, but its looks like its fixed. will remove.

File: client/src/components/MapMarker/__tests__/__snapshots__/AccuracyHalo.js.snap (hunk 3 above)
Comment (OSPFNeighbour) [reply client/src/components/MapMarker/__tests__/__snapshots__/AccuracyHalo.js.snap]: My spelling is just that bad

File: client/src/components/MapMarker/AccuracyHalo.js (hunk 2 above)
Comment (OSPFNeighbour) [reply client/src/components/MapMarker/AccuracyHalo.js]: was wonder how to do that

File: client/src/components/MapMarker/__tests__/__snapshots__/AccuracyHalo.js.snap (hunk 3 above)
Comment (sdunster) [reply client/src/components/MapMarker/__tests__/__snapshots__/AccuracyHalo.js.snap]: I dont see it

File: client/src/screens/events/Detail.js [hunk 4]
```diff
@@ -238,7 +238,7 @@ mapZoomMe = () => {
     const myLastPosition = this.state.myPosition;
     const myPosition = position.coords;
     // if we already have a high accuracy location ignore this fuzzy location
-    if (!HighAccuracy && !myLastPosition) {
+    if ((!HighAccuracy && !myLastPosition) || HighAccuracy) {
```
Comment (sdunster) [inline client/src/screens/events/Detail.js]: `highAccuracy` - change the function param to lower case

File: client/src/screens/events/Detail.js [hunk 5]
```diff
@@ -214,7 +214,7 @@ mapZoomMe = () => {
       return;
     }
     if (this.geoWatch == null) { // If this is the first run
-      // start a single fuzzy location fix aqusition
+    // start a single fuzzy location fix aqusition
```
[the remark this answers is not in the document]
Comment (sdunster) [reply client/src/screens/events/Detail.js]: maybe cause you also have a comment not he same line as the if statement? move that one before the `if`. Also while you are there `s/aqusition/acquisition` (i also had to consult dictionary)

File: client/src/components/MapMarker/__tests__/AccuracyHalo.js [hunk 6]
```diff
@@ -6,8 +6,7 @@ import AccuracyHalo from '../AccuracyHalo';
 const valid = {
   latitude: 1,
   longitude: 1,
-  heading: 1,
-  accurary: 1,
+  accuracy: 1,
```
[the remark this answers is not in the document]
Comment (sdunster) [reply client/src/components/MapMarker/__tests__/AccuracyHalo.js]: fair point, but it's good to write all your code twice so you can be really sure of every decision/keystroke :P

**Pull request merged**